### PR TITLE
fix(store): add migration for create selector generics

### PR DESCRIPTION
**Pull request by david-shortman**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## What is the new behavior?

Migrates createSelector generics

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/store/migrations/13_0_1/index.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+              ? `[${arg.map((a) => a.getText(sourceFile)).join(',')}]`
+              : arg.getText(sourceFile)
+          )
+          .join(',');
+
+        changes.push(
+          new RemoveChange(
+            sourceFile.fileName,
+            typeArguments.pos,
+            typeArguments?.end ?? 0
```
Comment (timdeschryver) [inline modules/store/migrations/13_0_1/index.ts:74]:
```suggestion
            typeArguments.end
```

File: modules/store/migrations/13_0_1/index.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        );
+        const updatedTypeArguments = [
+          typeArguments[0],
+          slicesTypeArguments,
+          typeArguments[lastTypeArgumentIndex],
+        ];
+
+        const isSelectorWithProps =
+          node.arguments.length === typeArguments.length - 2;
+        if (isSelectorWithProps) return;
```
Comment (timdeschryver) [inline modules/store/migrations/13_0_1/index.ts:60]: Should we log something for these cases?
I think the Angular migrations do this something, when a manual intervention is needed, which I liked.

Comment (david-shortman) [reply modules/store/migrations/13_0_1/index.ts:60]: The selectors with props require no migration because their signatures have all been left unmodified.

Comment (timdeschryver) [reply modules/store/migrations/13_0_1/index.ts:60]: Oooooooh, right... 😅

File: modules/store/migrations/migration.json [hunk 3]
```diff
@@ -20,6 +20,11 @@
       "description": "The road to v13 beta",
       "version": "13-beta",
       "factory": "./13_0_0-beta/index"
+    },
+    "ngrx-store-migration-13-0-1": {
+      "description": "The road to v13.0.1",
```
Comment (markostanimirovic) [inline modules/store/migrations/migration.json:25]: @david-shortman FYI, version `13.0.0` is not published yet. This migration should be included in `v13.0.0-rc.1` or `v13.0.0`.

cc @brandonroberts

**Review by timdeschryver — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**